blog/2015/10/teaching_extreme_ranges.html

```diff
@@ -0,0 +1,117 @@
+---
+layout: blog
+root: ../../..
+author: Daniel Chen
+title: "Teaching Bimodal Workshops with a High Range"
+date: 2015-10-31
+time: "09:00:00"
+category: ["Teaching"]
+---
+<!-- start excerpt -->
+
+<p>
+  Teaching is hard.
+  Teaching a technical class always has its challenges.
+  Teaching a technical workshop covering 3 or more topics over the span of 2 days is what makes us Software-Carpentry instructors.
+</p>
+<p>
+  A common observation instructors have is the class demographics are bimodal/multimodal.
+  There are students who attend the workshop with little to no experience,
+  some have been using the tools we teach but want a more formal course,
+  and finally, a student may know some of the tools, but use the workshop to learn git or python, for example.
+  How do we as instructors deal with workshops where the range of student skills is extremely high?
+</p>
+<p>
+  This post is a case-study of my most recent workshop and a call for help and discussion about we should do to make our workshops enjoyable for everyone.
+</p>
+
+<!-- end excerpt -->
+
+A few weeks ago I taught with Zhou Fu at
+<a href='http://zhuoaprilfu.github.io/2015-10-12-VirginiaTech/'>Virginia Tech</a>.
+We used the usual SWC low-tech-but-highly-effective dual-color sticky-note system,
+and at the end of each day or session we asked students to provide <a href='https://docs.google.com/document/d/1djwDkkaaZh2PcQtuehg9NeRznsdRynIdom3pB807-ec/edit?usp=sharing'>feedback</a>:
+something good, something bad.
+
+My co-instructor, Zhou, taught Bash and Git while I lead a
+<a href='https://github.com/swcarpentry/git-novice/issues/148'>Git capstone project</a>, and the Python sessions.
+It's great that our lessons are modular and self-contained,
+if a student gets lost in either of the Bash, Git, or Python lessons, it does not hinder their ability to learn the other topic
+(although being able to `cd` to a directory and knowing what a working directory does help).
+
+This workshop was different for me in that I've never had to teach a group with this wide of a skill gap.
+One one end, we had novices completely new to everything we had to offer in the workshop.
+On the other, we had a SysAdmin looking to transition Pearl scripts to Python.
```

Comment:
"looking to transition Pearl scripts to Python" -> "looking to port Pearl scripts into Python"?